public_html/wp-content/themes/pattern-directory/src/components/report-pattern-modal/index.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo, useReducer, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Button, Modal, RadioControl, Spinner, TextareaControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { store as patternStore } from '../../store';
+
+const reducer = ( state, action ) => {
+	switch ( action.status ) {
+		case 'submitting':
+			return { isSubmitting: true };
+		case 'submitted':
+			return { isSubmitted: true };
+		case 'error':
+			return { hasError: true, message: action.message };
+		default:
+			return {};
+	}
+};
+
+const ReportPatternModal = ( { postId, onClose } ) => {
+	const [ state, dispatch ] = useReducer( reducer, {} );
+	const [ selectedOption, setOption ] = useState( '' );
+	const [ details, setDetails ] = useState( '' );
+
+	const { isLoading, reasons } = useSelect( ( select ) => {
+		const { getPatternFlagReasons, isLoadingPatternFlagReasons } = select( patternStore );
+
+		return {
+			isLoading: isLoadingPatternFlagReasons(),
+			reasons: getPatternFlagReasons(),
+		};
+	} );
+
+	const handleSubmit = ( event ) => {
+		event.preventDefault();
+
+		if ( ! selectedOption || ! details.length || state.isSubmitted || state.isSubmitting ) {
+			return;
+		}
+
+		dispatch( { status: 'submitting' } );
+
+		apiFetch( {
+			path: addQueryArgs( '/wp/v2/wporg-pattern-flag' ),
+			method: 'POST',
+			data: {
+				parent: postId,
+				'wporg-pattern-flag-reason': selectedOption,
+				excerpt: details,
+			},
+		} )
+			.then( () => {
+				dispatch( { status: 'submitted' } );
```

Comment:
After submission, a screen reader user is left in limbo with no success message & focus on nothing, since the submit button disappears. You could add an announcement for success or error, using [wp.a11y.speak](https://github.com/WordPress/gutenberg/blob/cebcfb04281f105174303a453bc91cfa8ce3a136/packages/a11y/README.md#speak), and move focus back to the modal container (on success - error should be fine as-is, since the button stays on screen).

Reply:
`move focus back to the modal container`... Is there an ideal way to move the focus back?

Reply:
Usually I would say using `ref`s, but it seems that Modal doesn't pass refs down to the dialog container… I tried playing around with a few different ideas, but ended up back at refs with a wrapper div that would stay on screen through all states. I pushed my solution to this PR, but feel free to undo it if you want to look into another solution.

Reply:
No, it's fine. I was hoping there was a function in one of the WordPress packages to grab the ref for the `Modal`. 👍